modules/exploits/windows/http/pgadmin_binary_path_api.rb

```diff
@@ -0,0 +1,228 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'pgAdmin Binary Path API RCE',
+        'Description' => %q{
+          pgAdmin <= 8.4 is affected by a Remote Code Execution (RCE)
+          vulnerability through the validate binary path API. This vulnerability
+          allows attackers to execute arbitrary code on the server hosting PGAdmin,
+          posing a severe risk to the database management system's integrity and the security of the underlying data.
+
+          Tested on pgAdmin 8.4 on Windows 10 both authenticated and unauthenticated.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'M.Selim Karahan', # metasploit module
+          'Mustafa Mutlu', # lab prep. and QA
+          'Ayoub Mokhtar' # vulnerability discovery and write up
+        ],
+        'References' => [
+          [ 'CVE', '2024-3116'],
+          [ 'URL', 'https://ayoubmokhtar.com/post/remote_code_execution_pgadmin_8.4-cve-2024-3116/'],
+          [ 'URL', 'https://www.vicarius.io/vsociety/posts/remote-code-execution-vulnerability-in-pgadmin-cve-2024-3116']
+        ],
+        'Platform' => ['windows'],
+        'Arch' => ARCH_X64,
```

Comment:
Only x64, no x86 nor arm?

Reply:
![image](https://github.com/user-attachments/assets/d7234179-1ac4-4ebf-b8ff-cc01c7edd074)
Postgresql doesn't provide installers for x86 and arm. So I assumed there will no target which runs on these architectures